src/chainstate/stacks/miner.rs

```diff
@@ -1183,24 +1185,37 @@ impl StacksBlockBuilder {
         let parent_index_hash =
             StacksBlockHeader::make_index_block_hash(&parent_consensus_hash, &parent_header_hash);
 
-        let parent_microblocks = match self.load_parent_microblocks(
-            chainstate,
+        let burn_tip = SortitionDB::get_canonical_chain_tip_bhh(burn_dbconn.conn())?;
+        let burn_tip_height =
+            SortitionDB::get_canonical_burn_chain_tip(burn_dbconn.conn())?.block_height as u32;
+
+        let parent_microblocks = if StacksChainState::block_crosses_epoch_boundary(
+            chainstate.db(),
             &parent_consensus_hash,
             &parent_header_hash,
-            &parent_index_hash,
-        ) {
-            Ok(x) => x,
-            Err(e) => {
-                warn!("Miner failed to load parent microblock, mining without parent microblock tail";
-                      "parent_block_hash" => %parent_header_hash,
-                      "parent_index_hash" => %parent_header_hash,
-                      "parent_consensus_hash" => %parent_header_hash,
-                      "parent_microblock_hash" => match self.parent_microblock_hash.as_ref() {
-                          Some(x) => format!("Some({})", x.to_string()),
-                          None => "None".to_string(),
-                      },
-                      "error" => ?e);
-                vec![]
+        )? {
+            info!("Descendant of {}/{} will NOT confirm any microblocks, since it will cross an epoch boundary", &parent_consensus_hash, &parent_header_hash);
+            vec![]
+        } else {
+            match self.load_parent_microblocks(
+                chainstate,
+                &parent_consensus_hash,
+                &parent_header_hash,
+                &parent_index_hash,
+            ) {
+                Ok(x) => x,
+                Err(e) => {
+                    warn!("Miner failed to load parent microblock, mining without parent microblock tail";
+                              "parent_block_hash" => %parent_header_hash,
+                              "parent_index_hash" => %parent_header_hash,
+                              "parent_consensus_hash" => %parent_header_hash,
```

Comment:
These look like they were wrong before, but might as well correct them here -- I think two of these should be `%parent_consensus_hash` and `%parent_index_hash`.

Reply:
Haha wow. Fixed that.